src/core/model/StrokeStyle.cpp

```diff
@@ -1,91 +1,82 @@
-    PARSE_STYLE("dashdot", dashDotLinePattern);
-    PARSE_STYLE("dot", dotLinePattern);
+constexpr auto CUST_KEY = "cust: ";
 
+const std::map<std::string, std::vector<double>> predefinedPatterns = {
+    { "dash", {6, 3} }, 
+    { "dashdot", {6, 3, 0.5, 3} },
+    { "dot", {0.5, 3}}
+};
 
-    if (strncmp("cust: ", style, 6) != 0) {
-        return LineStyle();
-    }
+auto formatStyle(const std::vector<double>& dashes) -> std::string {
 
-    std::vector<double> dash;
-
-    const char* widths = style + 6;
-    while (*widths != 0) {
-        char* tmpptr = nullptr;
-        double val = g_ascii_strtod(widths, &tmpptr);
-        if (tmpptr == widths) {
-            break;
+    // Check if dashes match named predefined dashes.
+    for (auto &pair: predefinedPatterns ) {
+        if (pair.second == dashes) {
+            return pair.first;
         }
-        widths = tmpptr;
-        dash.push_back(val);
-    }
-
-    if (dash.empty()) {
-        return LineStyle();
     }
 
-    auto* dashesArr = new double[dash.size()];
-    for (int i = 0; i < static_cast<int>(dash.size()); i++) { dashesArr[i] = dash[i]; }
+    // Else generate custom dashes string
+    std::ostringstream custom;
+    custom << std::setprecision(2) << std::fixed;
+    custom << CUST_KEY;
+    std::copy(dashes.begin(), dashes.end(),std::ostream_iterator<double>(custom," "));
 
-    LineStyle ls;
-    ls.setDashes(dashesArr, static_cast<int>(dash.size()));
-    delete[] dashesArr;
+    // Return dashes string with traling space removed.
+    return custom.str().substr(0, custom.str().length() - 1);
+}
 
-    return ls;
 }
 
-#define FORMAT_STYLE(name, def)                                                                            \
-    if (count == (sizeof(def) / sizeof((def)[0])) && memcmp(dashes, def, count * sizeof((def)[0])) == 0) { \
-        return name;                                                                                       \
+auto StrokeStyle::parseStyle(const std::string &style) -> LineStyle {
+    auto it = predefinedPatterns.find(style);
+    if (it != predefinedPatterns.end()) {
+        LineStyle ls;
+        std::vector<double> dashes = it->second;
+        ls.setDashes(std::move(dashes));
+        return ls;
     }
 
-auto StrokeStyle::formatStyle(const double* dashes, int count) -> std::string {
-    FORMAT_STYLE("dash", dashLinePattern);
-    FORMAT_STYLE("dashdot", dashDotLinePattern);
-    FORMAT_STYLE("dot", dotLinePattern);
+    if (style.substr(0, strlen(CUST_KEY)) != CUST_KEY) {
+        return LineStyle();
+    }
 
-    std::string custom = "cust:";
+    std::stringstream dashStream(style);
+    std::vector<double> dashes;
 
-    for (int i = 0; i < count; i++) {
-        custom += " ";
-        char* str = g_strdup_printf("%0.2lf", dashes[i]);
-        custom += str;
-        g_free(str);
+    dashStream.seekg(strlen(CUST_KEY));
+    for (double value; dashStream >> value;)
+        dashes.push_back(value);
```

Comment:
```suggestion
    for (double value; dashStream >> value;) {
        dashes.push_back(value);
    }
```
See Coding conventions